src/librustdoc/json/mod.rs

```diff
@@ -189,6 +189,11 @@ impl<'tcx> FormatRenderer<'tcx> for JsonRenderer<'tcx> {
 
     fn after_krate(&mut self) -> Result<(), Error> {
         debug!("Done with crate");
+
+        for primitive in Rc::clone(&self.cache).primitive_locations.values() {
```

Comment:
```suggestion
        for primitive in self.cache.primitive_locations.values() {
```

Reply:
I don't think the change compiles. `.values()` immutably borrows `self`, and `.get_impls()` tries to mutably borrow `self` while the immutable reference is alive.